Gems/AWSCore/Code/Platform/Linux/AWSCore_Traits_Linux.h

```diff
@@ -7,4 +7,4 @@
  */
 #pragma once
 
-#define AWSCORE_BACKWARD_INCOMPATIBLE_CHANGE 0
+#define AWSCORE_BACKWARD_INCOMPATIBLE_CHANGE 1
```

Comment:
nit: Can you talk to this a little, provide some context as to what this flag means and the state of things after the update?

Reply:
Change is related to https://github.com/o3de/o3de/blob/a486d1a8ad7fa349313fd2742fa762c347d21c88/Gems/AWSCore/Code/Include/Framework/AWSApiJobConfig.h#L139-L143
AWS CPP SDK has made backward incompatible change which changes a bool field to FollowRedirectsPolicy, once all platform have been updated to use latest sdk, we can get rid of this flag